### PR TITLE
Sieve of erastothenes || documentation || mathematics

### LINKED ISSUE #303: documentation for sieve of eratosthenes || math
## 🚀 Feature

documentation for sieve of eratosthenes

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

documentation for sieve of eratosthenes

## Pitch

documentation for sieve of eratosthenes

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #303 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #303

### CODE REVIEW

File: docs/mathematics/sieve-of-eratosthenes.mdx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+---
+
+## Introduction
+
+The sieve of Eratosthenes is an algorithm for finding all prime numbers up to any given limit. It is computationally highly efficient algorithm.
+
+## Explanation
+
+Let us take the number `n = 100` , so we create list from `2` to `100` . According to the algorithm , we mark all the numbers which are divisible by `2` which are greater than or equal to the square of it . Then we move to the next number `3` , we mark all the numbers which are divisible by `3`  which are greater than or equal to the square of it. we continue this process until the square of the number is greater than `n`. The numbers which are unmarked in the list are the prime numbers less than `n`.
+
```
Comment (atarax665) [inline docs/mathematics/sieve-of-eratosthenes.mdx:13]: If would be great if you could add some pictures to explain the workflow of the algorithm.

File: docs/mathematics/sieve-of-eratosthenes.mdx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+void print(int N, bool *isprime)
+{
+    for (int i = 1; i <= N; i++)
+    {
+        if (!isprime[i])
+        {
+            std::cout << i << ' ';
+        }
+    }
+    std::cout << '\n';
```
Comment (atarax665) [inline docs/mathematics/sieve-of-eratosthenes.mdx:84]:
```suggestion
    cout << '\n';
```

File: docs/mathematics/sieve-of-eratosthenes.mdx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+#include <iostream>  //Header file
```
Comment (atarax665) [inline docs/mathematics/sieve-of-eratosthenes.mdx:39]: Remove trivial comments.

File: docs/mathematics/sieve-of-eratosthenes.mdx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+## Complexity Analysis
+
+- Time complexity - O(`n*(log(log(n)))`)
+- Space complexity - O(`n`)
+
+
+## Credits
+
+-   [Rajiv Singh](https://github.com/iamrajiv) for [implementation in c++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/Sieve_of_Eratosthenes.cpp)
+-   [Adithya kurien peter](https://github.com/aaadddiii) for implementation in python
```
Comment (atarax665) [inline docs/mathematics/sieve-of-eratosthenes.mdx:171]: You haven't linked the python code.

File: docs/mathematics/sieve-of-eratosthenes.mdx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+## Complexity Analysis
+
+- Time complexity - O(`n*(log(log(n)))`)
+- Space complexity - O(`n`)
+
+
+## Credits
+
+-   [Rajiv Singh](https://github.com/iamrajiv) for [implementation in c++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/Sieve_of_Eratosthenes.cpp)
```
Comment (atarax665) [inline docs/mathematics/sieve-of-eratosthenes.mdx:170]:
```suggestion
-   [Rajiv Singh](https://github.com/iamrajiv) for [implementation in C++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/Sieve_of_Eratosthenes.cpp)
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@atarax665  I have made the changes can you please review it

**Review by atarax665 — APPROVED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  Can you please review and merge my pr

**Comment by aaadddiii** [comment]
@iamrajiv  can you please review my pr

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**